src/prefect/cli/block.py

```diff
@@ -109,3 +112,58 @@ async def register(
         "\n To configure the newly registered blocks, "
         "go to the Blocks page in the Prefect UI.\n"
     )
+
+
+@blocks_app.command()
+async def ls():
+    """
+    View all configured Blocks.
+    """
+    async with get_client() as client:
+        blocks = await client.read_block_documents()
+
+    table = Table(
+        title="Blocks", caption="List Block Types using `prefect block type ls`"
+    )
+    table.add_column("ID", style="cyan", no_wrap=True)
+    table.add_column("Type", style="blue", no_wrap=True)
+    table.add_column("Name", style="blue", no_wrap=True)
+    table.add_column("Slug", style="blue", no_wrap=True)
+
+    for block in sorted(blocks, key=lambda x: x.block_schema.fields["title"]):
```

Comment:
I think that sorting by block slug would work well since it should group by block type first and then alphabetical by block name. Maybe also a function to get the block slug keep things DRY.
```suggestion
    for block in sorted(blocks, key=lambda x: f"{x.block_type.slug}/{x.name}"):
```